product_variant_configurator/models/pricelist.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016 ACSONE SA/NV
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from odoo import api, models
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    @api.multi
+    def _compute_price_rule(self, products_qty_partner, date=False,
+                            uom_id=False):
+        """Overwrite for covering the case where templates are passed and a
+        different uom is used."""
+        if products_qty_partner[0][0]._name != "product.template":
+            # Standard use case - Nothing to do
+            return super(ProductPricelist, self)._compute_price_rule(
+                products_qty_partner, date=date, uom_id=uom_id,
+            )
+        if not uom_id and self.env.context.get('uom'):
+            uom_id = self.env.context['uom']
+            ctx = dict(self.env.context)
+            ctx.pop('uom')  # Remove uom context for avoiding the re-processing
```

Comment:
You can do directly (my fault): `uom_id = ctx.pop('uom')` and save the first line